File: frontend/src/shared/schemas/personnelExtra.ts
import { z } from "zod";

import { adminPositionSchema, type AdminPosition } from "./enums";

// schema ของ sub-resource บุคลากร — ต้องตรงกับ backend (service.AdminPositionDTO / AcademicStandingDTO)

// label ตำแหน่งบริหาร (ใช้ enum กลางจาก enums.ts)
export const positionLabel: Record<AdminPosition, string> = {
  director: "ผู้อำนวยการ",
  deputy_director: "รองผู้อำนวยการ",
};

// --- ตำแหน่งบริหาร ---
export const adminPositionRecordSchema = z.object({
  id: z.string(),
  position: z.string(),
  is_active: z.boolean(),
  appointed_at: z.string(),
  created_at: z.string(),
});
export type AdminPositionRecord = z.infer<typeof adminPositionRecordSchema>;
export const adminPositionListSchema = z.array(adminPositionRecordSchema);

export type CreatePositionBody = {
  position: AdminPosition;
  appointed_at: string;
};

// --- วิทยฐานะ ---
export const academicStandingRecordSchema = z.object({
  id: z.string(),
  standing: z.string(),
  effective_date: z.string(),
  is_current: z.boolean(),
  created_at: z.string(),
});
export type AcademicStandingRecord = z.infer<typeof academicStandingRecordSchema>;
export const academicStandingListSchema = z.array(academicStandingRecordSchema);

export type StandingBody = {
  standing: string;
  effective_date: string;
  is_current: boolean;
};

// re-export enum สำหรับใช้ใน form
export { adminPositionSchema };

File: backend/internal/domain/personnel_extra.go
package domain

import "time"

// ตำแหน่งบริหาร
const (
	PositionDirector       = "director"
	PositionDeputyDirector = "deputy_director"
)

// AdminPosition คือตำแหน่งบริหารของบุคลากร (ผอ./รอง ผอ.)
type AdminPosition struct {
	ID          string
	SchoolID    string
	PersonnelID string
	Position    string
	IsActive    bool
	AppointedAt *time.Time
	CreatedAt   time.Time
	UpdatedAt   time.Time
}

// NewAdminPosition payload สร้างตำแหน่งบริหาร
type NewAdminPosition struct {
	Position    string
	IsActive    bool
	AppointedAt *time.Time
}

// AcademicStanding คือวิทยฐานะ (เก็บเป็นประวัติหลายรายการ; is_current ได้แค่ 1 ต่อคน)
type AcademicStanding struct {
	ID            string
	SchoolID      string
	PersonnelID   string
	Standing      string
	EffectiveDate *time.Time
	IsCurrent     bool
	CreatedAt     time.Time
	UpdatedAt     time.Time
}

// NewAcademicStanding payload สร้างวิทยฐานะ
type NewAcademicStanding struct {
	Standing      string
	EffectiveDate *time.Time
	IsCurrent     bool
}

// UpdateAcademicStanding payload แก้ไขวิทยฐานะ
type UpdateAcademicStanding struct {
	Standing      string
	EffectiveDate *time.Time
	IsCurrent     bool
}
